tests/multi_objective/test_trial.py

```diff
@@ -0,0 +1,203 @@
+    assert isinstance(study.trials[0].datetime_complete, datetime.datetime)
+
+
+def test_dominates() -> None:
+    directions = [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE]
+
+    def create_trial(
+        values: List[Optional[float]], state: TrialState = TrialState.COMPLETE,
+    ) -> FrozenMultiObjectiveTrial:
+        n_objectives = len(values)
+        trial = optuna.structs.FrozenTrial(
+            state=state,
+            intermediate_values={i: v for i, v in enumerate(values)},
+            # The following attributes aren't used in this test case.
+            number=0,
+            value=None,
+            datetime_start=None,
+            datetime_complete=None,
+            params={},
+            distributions={},
+            user_attrs={},
+            system_attrs={},
+            trial_id=0,
+        )
+        return FrozenMultiObjectiveTrial(n_objectives, trial)
+
+    # The numbers of objectives for `t0` and `t1` don't match.
+    with pytest.raises(ValueError):
+        t0 = create_trial([1])  # One objective.
+        t1 = create_trial([1, 2])  # Two objectives.
+        t0._dominates(t1, directions)
+
+    # The numbers of objectives and directions don't match.
+    with pytest.raises(ValueError):
+        t0 = create_trial([1])  # One objective.
+        t1 = create_trial([1])  # One objective.
+        t0._dominates(t1, directions)
+
+    # `t0` dominates `t1`.
+    t0 = create_trial([0, 2])
+    t1 = create_trial([1, 1])
+    assert t0._dominates(t1, directions)
+    assert not t1._dominates(t0, directions)
+
+    # `t0` dominates `t1`.
+    t0 = create_trial([0, 1])
+    t1 = create_trial([1, 1])
+    assert t0._dominates(t1, directions)
+    assert not t1._dominates(t0, directions)
+
+    # `t0` and `t1` don't dominate each other.
+    t0 = create_trial([1, 1])
+    t1 = create_trial([1, 1])
+    assert not t0._dominates(t1, directions)
+    assert not t1._dominates(t0, directions)
+
+    # `t0` and `t1` don't dominate each other.
+    t0 = create_trial([0, 1])
+    t1 = create_trial([1, 2])
+    assert not t0._dominates(t1, directions)
+    assert not t1._dominates(t0, directions)
+
+    for t0_state in [TrialState.FAIL, TrialState.WAITING, TrialState.PRUNED]:
+        t0 = create_trial([1, 1], t0_state)
+
+        for t1_state in [
+            TrialState.COMPLETE,
+            TrialState.FAIL,
+            TrialState.WAITING,
+            TrialState.PRUNED,
+        ]:
+            # If `t0` has not the COMPLETE state, it never dominates other trials.
+            t1 = create_trial([0, 2], t1_state)
+            assert not t0._dominates(t1, directions)
+
+            if t1_state == TrialState.COMPLETE:
+                # If `t0` isn't COMPLETE and `t1` is COMPLETE, `t1` dominates `t0`.
+                assert t1._dominates(t0, directions)
+            else:
+                # If both `t0` and `t1` aren't COMPLETE, they don't dominate each other.
```

Comment:
This comment should be `# If both `t1` isn't COMPLETE, it doesn't dominate others.`

Reply:
Thanks. Fixed: https://github.com/optuna/optuna/pull/1054/commits/05473323a1bcaf2e1c15fc8da75d22ad6975fbef